### ISSUE TITLE
VoodooPS2 issues with Synaptics/PS2 touchpad

### ISSUE THREAD

**Issue by Tarik02** [acidanthera/bugtracker#616]
I have a `Synaptics/PS2` touchpad on Lenovo s510p. It works normally with RehabMan's VoodooPS2Controller, seems to work ok with ApplePS2Controller, but has very strange troubles with your's VoodooPS2. It seems to work ok with mouse and keyboard, but when I try to use touchpad, here issues go: when I touch touchpad with single finger, it moves (but not so smoothly as it was with RehabMan's, something like it is some jumpy when it is moved slowly); but when I touch with 2 or more fingers (this touchpad supports 2 fingers tracking and up to 3 counting), touchpad stops working at all.

P.S. Touchpad works ok with Linux's libinput, xf86-synaptics and mtrack. On Windows I could even install Microsoft Precision Touchpad Driver (but with issues) on Windows.

**Comment by Tarik02** [comment]
Seems like it reports events, but it can't get coordinates of second finger (they're always same).
![Знімок екрана 2019-12-21 о 14 53 40](https://user-images.githubusercontent.com/12175048/71308169-d24c3b80-2401-11ea-9e1c-76de34fbbf9b.png)

However, when I move three fingers, first two are reported, but the third is not.
![Знімок екрана 2019-12-21 о 14 56 24](https://user-images.githubusercontent.com/12175048/71308184-10e1f600-2402-11ea-8600-7bfb5db5348c.png)

While this all, I can't get touchpad working. Events in console go, but pointer does not move at all (neither clicks with both touchpad and the buttons behind touchpad).

**Comment by Tarik02** [comment]
All kexts are loaded correctly.
![Знімок екрана 2019-12-21 о 15 05 22](https://user-images.githubusercontent.com/12175048/71308296-518e3f00-2403-11ea-98a7-50f2baf110d3.png)

**Comment by Tarik02** [comment]
Can't get it working at least somehow anymore. Neither debug nor release, neither my compiled build or build from releases. Kext loaded, logs are written, but cursor does not move.

One more interesting fact is that `fingerStates[1].x` and `fingerStates[1].y` never gets changed (always `0, 0`) no matter two or three fingers on the touchpad (they can be changed from extended packet, but it never gets received).

**Comment by Tarik02** [comment]
So. I tried different versions and here's my results:

```
2.0.0.1 - does not detect touchpad
2.0.2 - works, strange moving
2.0.3 - gestures works very good (but strangely at start); does not detect any two-finger gesture (including two finger click and scrolling)
2.0.4 - same as 2.0.3
2.1.0 - does not work
```

There's something strange with 2.1.0 with VoodooInput. I installed it, rebooted and tried to move the cursor, it moved. As soon as I touch the touchpad with two fingers, it stops working at all. It does not work even after reboot! Then I reverted to 2.0.4 (of course, removed VoodooInput), touchpad worked. But when I installed 2.1.0 with VoodooInput, it does not work at all.

Soo... there's some problem either with VoodooInput or 2.1.0 or connection between VoodooInput and 2.1.0.
